### ISSUE TITLE
Link configuration for reporter.testId and reporter.issue API's

### ISSUE THREAD

**Issue by KiT-Maverik** [webdriverio/wdio-allure-reporter#158]
Hey everybody!
I run into a little problem and I hope you can help me solve it.

FRAMEWORK: 
	WDIO
	MOCHA
	CHAI
	WDIO-ALLURE-REPORTER

When I'm trying to assign test case ID (via reporter.testId(999)) or issue ID (via reporter.issue(111)) the reporter successfully adds corresponding information to Allure report. But when I click the link it navigates me to Allure server IP. Link looks like the folloing example:
http://10.11.12.123:24240/999  (for reporter.testId(999))

It it very clear for that there should be config property somewhere, intended to regulate this, but I didn't managed to find any information.
Please assist!

**Comment by BorisOsipov** [comment]
And againg it is not about wdio-reporter :)

>I didn't managed to find any information.

It is in allure docs. See https://docs.qameta.io/allure/#_links

**Comment by KiT-Maverik** [comment]
@BorisOsipov 
Thanks!
Okay, I am about to went to bug smb on Allure side) 
One more little question. that pattern - related staff. Is it supported by wdio allure reporter? Or this is something to be handled on allure core side?
it appears to me that there should be somkind of API in reporter objet allowing set pattern.

**Comment by BorisOsipov** [comment]
>it appears to me that there should be somkind of API in reporter objet allowing set pattern.


@baev can you assist? rigth now reporter just add `<label name='issue' value='gh-1212'/>` to testcase. How can we pass issue\link pattern?

**Comment by baev** [comment]
You can create `allure.properties` file in results directory and use properties `allure.issues.tracker.pattern` and `allure.tests.management.pattern` for patterns, for example:

```properties
allure.issues.tracker.pattern: https://example.org/issue/%s
allure.tests.management.pattern: https://example.org/tms/%s
```

Or you can use full links directly (providing label `<label name='issue' value='https://github.com/some/repo/1212'/>`). 

This is only applied for results in Allure 1 XML format. In new model we moved pattren resolution logic to adaptor side (it is easier for users to configure and in that case each result file can be processed individually that required for realtime reporting)

**Comment by KiT-Maverik** [comment]
@baev @BorisOsipov 
Thanks guys! It works.
But from my perspective it would be nice if only we could manage these properties from reporter config or something.
Anyway, thanks!

**Comment by baev** [comment]
> But from my perspective it would be nice if only we could manage these properties from reporter config or something.

Yep, I agree. We are already working on new JS integrations https://github.com/allure-framework/allure-js/
